django/researchdata/models.py

```diff
@@ -0,0 +1,332 @@
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        abstract = True
+        ordering = [Upper('name'), 'id']
+
+
+class SlEntityType(SlAbstract):
+    """
+    Select List model used by Entity model (inherits from SlAbstract model)
+    The type/category of an entity. E.g. publisher, university, department, centre
+    """
+
+
+class SlEventActivity(SlAbstract):
+    """
+    Select List model used by Event model (inherits from SlAbstract model)
+    The activity that's being carried out at the event. E.g. publishing, writing, speaking
+    """
+
+
+class SlEventType(SlAbstract):
+    """
+    Select List model used by Event model (inherits from SlAbstract model)
+    The type/category of an event. E.g. conference, seminar, meeting
+    """
+
+
+class SlItemMedia(SlAbstract):
+    """
+    Select List model used by Item model (inherits from SlAbstract model)
+    The type of media of an item. E.g. print, video
+    """
+
+
+class SlItemType(SlAbstract):
+    """
+    Select List model used by Item model (inherits from SlAbstract model)
+    The type/category of an item. E.g. minutes, memo, teaching materials
+    """
+
+
+class SlLanguage(SlAbstract):
+    """
+    Select List model used by various models (inherits from SlAbstract model)
+    A language. E.g. English, French, German
+    """
+
+
+class SlLocation(SlAbstract):
+    """
+    Select List model used by various models (inherits from SlAbstract model)
+    A geographical location. E.g. Birmingham, London, Paris
+    """
+
+
+class SlPersonTitle(SlAbstract):
+    """
+    Select List model used by Person model (inherits from SlAbstract model)
+    A title of a person, e.g. Mrs, Ms, Miss, Dr, Mr, etc.
+    """
+
+
+class SlTypeRelEntityAndEvent(SlAbstract):
+    """
+    Select List model used by RelEntityAndEvent (inherits from SlAbstract model)
+    A type of relationship between Entity and Event
+    """
+
+
+class SlTypeRelEntityAndItem(SlAbstract):
+    """
+    Select List model used by RelEntityAndItem (inherits from SlAbstract model)
+    A type of relationship between Entity and Item
+    """
+
+
+class SlTypeRelEntityAndPerson(SlAbstract):
+    """
+    Select List model used by RelEntityAndPerson (inherits from SlAbstract model)
+    A type of relationship between Entity and Person
+    """
+
+
+class SlTypeRelEventAndPerson(SlAbstract):
+    """
+    Select List model used by RelEventAndPerson (inherits from SlAbstract model)
+    A type of relationship between Event and Person
+    """
+
+
+class SlTypeRelItemAndItem(SlAbstract):
+    """
+    Select List model used by RelItemAndItem (inherits from SlAbstract model)
+    A type of relationship between Item and Item
+    """
+
+
+class SlTypeRelItemAndPerson(SlAbstract):
+    """
+    Select List model used by RelItemAndPerson (inherits from SlAbstract model)
+    A type of relationship between Item and Person
+    """
+
+
+# Main tables
+
+
+class Entity(models.Model):
```

Comment:
I would tend to shy away from naming a model "Entity", just because Entity has the specific meaning of ORM Entity/Model. However, I know it is difficult to come up with anything better. I have resorted to the very ugly 'BusinessUnit' for a project (admittedly in Java where the models are called entities). This is just an observation, I don't need you to justify the name.

Reply:
Yeah, it's not the best term, but it's what the researcher named it (though she admitted it was a big vague).